### PR TITLE
feat: Tab Item support icon prop

### LINKED ISSUE #46076: Tab Item support icon property
```jsx
  <Tabs
    defaultActiveKey="2"
    items={[
      { label: 'tab', icon: <AppleOutlined /> },
      { label: 'tab', icon: <AppleOutlined /> },
      { label: 'tab', icon: <AppleOutlined /> },
    ]}
  />
```

https://ant.design/components/tabs-cn#tabs-demo-icon

### PR DESCRIPTION
**Pull request by li-jia-nan**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

- close #46076

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | feat: Tab Item support icon prop |
| 🇨🇳 Chinese | Tab Item 支持 icon 属性 |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖[[deprecated]](https://githubnext.com/copilot-for-prs-sunset) Generated by Copilot at 76c76ae</samp>

Added the `icon` property to the `Tabs` component and its documentation. This allows users to specify icons for each tab head and improves the icon alignment and spacing. Updated the `rc-tabs` dependency to enable this feature.

### <samp>🤖[[deprecated]](https://githubnext.com/copilot-for-prs-sunset) Generated by Copilot at 76c76ae</samp>

*  Add `icon` property to `TabItemType` interface and `Tabs` component to display icons on tab heads ([link](https://github.com/ant-design/ant-design/pull/46096/files?diff=unified&w=0#diff-fabce1b872eec72221eb71183a6acf00a1ee22a5c224285991df2b3adb94730fL10-R14), [link](https://github.com/ant-design/ant-design/pull/46096/files?diff=unified&w=0#diff-55d7b2e14d2f70c11368544596a23fa53e1c2d392b85295e52cbf9d21bec7ebcR85),  [link](https://github.com/ant-design/ant-design/pull/46096/files?diff=unified&w=0#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L154-R154))
*  Update `genTabStyle` function to add margin to icon element for better spacing and alignment ([link](https://github.com/ant-design/ant-design/pull/46096/files?diff=unified&w=0#diff-d2d317a91ba62030532018942e2f2e91ffc533b238fb8017b7bea62d4ac428a7R694-R696))
*  Update `components/tabs/demo/icon.tsx` to use `icon` property instead of `label` for icons ([link](https://github.com/ant-design/ant-design/pull/46096/files?diff=unified&w=0#diff-fabce1b872eec72221eb71183a6acf00a1ee22a5c224285991df2b3adb94730fL10-R14))

### CODE REVIEW

File: components/tabs/style/index.ts [hunk 1]
```diff
@@ -691,6 +691,9 @@ const genTabStyle: GenerateStyle<TabsToken, CSSObject> = (token: TabsToken) => {
       '&-btn': {
         outline: 'none',
         transition: 'all 0.3s',
+        [`${tabCls}-icon`]: {
+          marginInlineEnd: token.marginXXS,
+        },
```
Comment (MadCcc) [inline components/tabs/style/index.ts:696]: 考虑下只有 icon 的情况？

Comment (li-jia-nan) [reply components/tabs/style/index.ts:696]: - https://github.com/react-component/tabs/pull/681

File: components/tabs/style/index.ts [hunk 2]
```diff
@@ -691,6 +691,12 @@ const genTabStyle: GenerateStyle<TabsToken, CSSObject> = (token: TabsToken) => {
       '&-btn': {
         outline: 'none',
         transition: 'all 0.3s',
+        [`${tabCls}-icon`]: {
```
Comment (MadCcc) [inline components/tabs/style/index.ts:694]:
```suggestion
        [`${tabCls}-icon:not(:last-child)`]: {
```

File: components/tabs/style/index.ts [hunk 3]
```diff
@@ -691,6 +691,12 @@ const genTabStyle: GenerateStyle<TabsToken, CSSObject> = (token: TabsToken) => {
...
+          marginInlineEnd: token.marginSM,
+          '&:only-child': {
+            marginInlineEnd: 0,
+          },
```
Comment (MadCcc) [inline components/tabs/style/index.ts:698]:
```suggestion
```

**Review by MadCcc — APPROVED** [review]

File: components/tabs/style/index.ts [hunk 4]
```diff
@@ -740,7 +740,7 @@ const genTabStyle: GenerateStyle<TabsToken, CSSObject> = (token: TabsToken) => {
       [`& ${tabCls}-remove ${iconCls}`]: {
         margin: 0,
       },
-      [iconCls]: {
+      [`${iconCls}:not(:last-child)`]: {
         marginRight: {
```
Comment (ANM699) [inline components/tabs/style/index.ts:741]: 貌似这里会导致antdpro在开启多标签模式时，icon的margin-right丢失

**Pull request merged**